### DISCUSSION TITLE
What type for monetary values

### DISCUSSION THREAD

**Discussion by dawidmachon** [payloadcms/payload#1958] [Q&A] [answered]
I would like to keep monetary (money/prices) in collections. Which type will be okay, number? But do I not lose precision on arithmetic problems?

Can I create own implementation for field compliant with: https://www.mongodb.com/docs/manual/tutorial/model-monetary-data/ ?

**Comment by JarrodMFlesch** [comment, marked as answer]
@dawidmachon It is generally best practice to store monetary values in the smallest denomination, there are plenty of good articles out there explaining why this is the case.

So personally I would opt to use the number field almost always (storing as cents), and then add a description that displays the amount in a dollar amount.

**Reply by jacobsfletch** [reply]
Related: https://github.com/payloadcms/payload/pull/1924

**Comment by MatthiasGrandl** [comment]
Would be nice to get a full on money field like https://craftcms.com/docs/4.x/money-fields.html
